Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -918,6 +918,48 @@ void NodeWithStyle::apply_style(CSS::ComputedProperties const& computed_style)
             computed_values.set_aspect_ratio({ false, aspect_ratio.as_ratio().ratio() });
     }
 
+    auto const& touch_action = computed_style.property(CSS::PropertyID::TouchAction);
+    if (touch_action.is_value_list())
+    {
+        CSS::TouchActionData touch_action_data = CSS::TouchActionData::none();
+        for (auto const& value : touch_action.as_value_list().values()) {
+            switch (value->as_keyword().keyword()) {
+            case CSS::Keyword::Auto:
+                touch_action_data = {};
+                break;
+            case CSS::Keyword::None:
+                break;
+            case CSS::Keyword::Manipulation:
+                touch_action_data = {};
+                touch_action_data.allow_other = false;
+                break;
+            case CSS::Keyword::PanX:
+                touch_action_data.allow_right = true;
+                touch_action_data.allow_left = true;
+                break;
+            case CSS::Keyword::PanLeft:
+                touch_action_data.allow_left = true;
+                break;
+            case CSS::Keyword::PanRight:
+                touch_action_data.allow_right = true;
+                break;
+            case CSS::Keyword::PanY:
+                touch_action_data.allow_up = true;
+                touch_action_data.allow_down = true;
+                break;
+            case CSS::Keyword::PanUp:
+                touch_action_data.allow_up = true;
+                break;
+            case CSS::Keyword::PanDown:
+                touch_action_data.allow_down = true;
+                break;
+            default:
+                VERIFY_NOT_REACHED();
+            }
+        }
+        computed_values.set_touch_action(touch_action_data);
+    }
```

Comment:
This code to compute the `TouchActionData` should go in a `ComputedProperties::touch_action()` method, and then used here.

Reply:
This makes sense. I thought that might be the right way but I was a bit hesitant because I saw that most of the ones doing significant computation were right in node but that's probably just because they haven't been moved over to ComputedProperties.

Would it be useful to move more of the business logic out of `NodeWithStyle::apply_style` (separate PR)?

Reply:
Sure, `apply_style()` has got unwieldy so making it less so would be good.